### PR TITLE
clean up SDK test fixtures

### PR DESCRIPTION
**Pull request by mpobrien**
this eliminates raciness from the tests, reenables parallel execution, refactors the SDK tests significantly, and adds EVG pull request hooks.

### CODE REVIEW

File: dist/web/stitch.js [hunk 1]
```diff
@@ -102,11 +102,14 @@ function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj;
 
 var RESULT_METADATA_KEY = '_stitch_metadata';
 
+/** @namespace util */
```
[the remark this answers is not in the document]
Comment (mpobrien) [reply dist/web/stitch.js]: These are here from a rebase on jason's code - it's already in the repo, so you can ignore these lines from the diff.

File: .evg.yml [hunk 2]
```diff
@@ -18,16 +18,29 @@ functions:
...
-          ./bin/mongo --nodb --eval 'assert.soon(function(x){try{var d = new Mongo("localhost:27017"); return true}catch(e){return false}}, "timed out connecting")'
+          ./bin/mongo --nodb --eval 'assert.soon(function(x){try{var d = new Mongo("localhost:26000"); return true}catch(e){return false}}, "timed out connecting")'
           echo "mongod is up."
 
+  "set_github_build_status":
+    - command: shell.exec
+      params:
+        silent: true
+        script: |
+          status_url=`curl -X POST -d "{}" -H "Content-Type: application/json" "${stitch_evg_hook_url}?secret=${stitch_evg_secret}&versionId=${version_id}"`
```
[the remark this answers is not in the document]
Comment (mpobrien) [reply .evg.yml]: Yes, good point, totally forgot about that. Maybe we should restrict who's allowed to send pull requests, if github allows that? I wouldn't feel comfortable about leaving this wide open, but it's too convenient to the workflow to take away.

[the remark this answers is not in the document]
Comment (edaniels) [reply .evg.yml]: We can make it work via stitch with a match stage and a value doing a %in

File: dist/web/stitch.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (edaniels) [reply dist/web/stitch.js]: Gotcha

File: dist/node/auth/providers.js [hunk 3]
```diff
@@ -60,11 +68,14 @@ function anonProvider(auth) {
   };
 }
 
+/** @namespace */
```
[the remark this answers is not in the document]
Comment (mpobrien) [reply dist/node/auth/providers.js]: 👍

File: .evg.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mpobrien) [reply .evg.yml]: 👍

**Pull request merged**